### DISCUSSION TITLE
Get Oracle Erro Code using SQLx and  go-ora

### DISCUSSION THREAD

**Discussion by amichelins** [sijms/go-ora#237] [General]
Im using go-ora in my developement
The package impressed me.

Now i am working whith Store Procedures, and i need to know the oracle erro code that is throw in the procedure.
this is my code.

This is my code. I try this and not give me result

`
                _, err = txw.Exec(sSql)

                if err != nil {
                    txw.Rollback()

                    if e, ok := err.(interface {
                        Code() int
                        Error() string
                    }); ok {
                        if e.Code() == -20020 {
                            Do work.....
                        } else {
                            Other Work
                        }
                    }

                }
`
Can you help me please ?

Thanks in advance

**Comment by sijms** [comment]
not all error returned as oracle error interface
for your code to work properly all errors should return as OracleError
I will check this and update the package

**Reply by amichelins** [reply]
Thanks to Add Oracle Erros. I am currently using v2.6.12 
Now, I have how to get Code and Message separatly. 
Your package help me alot, because i not depend any more fore C linking.
Build for Linux is direct from my Windows 10

**Comment by amichelins** [comment]
A this time the err variavel has this ( taken with debug ) has :

error(*github.com/sijms/go-ora/v2/network.OracleError) *{ErrCode: 6550, ErrMsg: "ORA-06550: line 1, column 27:\nPL/SQL: ORA-00942: table or view does not exist\nORA-06550: line 1, column 15:\nPL/SQL: SQL Statement ignored\n"}

This code is not working to catch the values:
```
          if e, ok := err.(interface {
               ErrCode() int
               ErrMsg() string
           }); ok {
               if e.Code() == 1 {
                   return Message here
               }
           }
```

How i transform the code tu get ErrCode and ErrMsg ?

**Reply by amichelins** [reply]
I try this code:
```
gooranet "github.com/sijms/go-ora/v2/network"
...
...
...
e, ok := err.(*gooranet.OracleError )

                if ok {
                    println( fmt.Sprintf("%d", e.ErrCode) )
                    println( fmt.Sprintf("%s", e.ErrMsg) )
                }
```
And worked fine

All go-ora errors are returned as OracleError ?

**Comment by sijms** [comment]
all oracle related errors are returned as OracleError

**Reply by amichelins** [reply]
thanks alot!!!
